clientconn.go

```diff
@@ -208,7 +208,11 @@ func newClient(target string, opts ...DialOption) (conn *ClientConn, err error)
 // https://github.com/grpc/grpc/blob/master/doc/naming.md.
 // e.g. to use dns resolver, a "dns:///" prefix should be applied to the target.
 func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *ClientConn, err error) {
-	cc, err := newClient(target, opts...)
+	// At the end of this method, we kick the channel out of idle, rather than waiting for the first rpc.
+	// The eagerConnect option is used to tell the ClientChannel that DialContext was called to make
+	// this channel, and hence tries to connect immediately.
+	opts = append(opts, WithEagerConnect())
+	cc, err := NewClient(target, opts...)
```

Comment:
Something like this could work, though I'd want to not export `WithEagerConnect`.

Another option would be:

```go
func newClient(target, defaultScheme string, opts ...DialOption) {}

func NewClient(..) (..) {
	return newClient(target, "dns", opts...)
}

func DialContext(..) (..) {
	cc, err := newClient(target, "passthrough", opts...)
	...
}
```

We could also combine the two ideas and make a `DialOption` that specifies the default scheme, and if it isn't specified, then we assume "dns" (and `DialContext` sets it with `"passthrough"`).

Reply:
Yep I like this, looks clean